pkg/auth/inject.go

```diff
@@ -42,9 +42,13 @@ func (m DependencyMap) Provide(dependencyName string, r *http.Request) interface
 	case "AuthInfoStore":
 		tConfig := config.GetTenantConfig(r)
 		return coreAuth.NewDefaultAuthInfoStore(r.Context(), tConfig)
+	case "AuthDataKeys":
```

Comment:
I think directly injecting the config into the handler is violating the current injection pattern.

I can think of two approaches:
- put the `ToValidAuthDataList` function in `AuthDataChecker` (and maybe give it a different name), OR
- merge `AuthDataChecker` into `PasswordAuthProvider` and put all auth record keys logic in it, because imo the handler does not even have to know how to process the given auth data